pkg/observer/observer.go

```diff
@@ -107,84 +89,7 @@ func (k *Observer) receiveEvent(data []byte, cpu int) {
 	}
 }
 
-func (k *Observer) __runEvents(stopCtx context.Context) (*bpf.PerCpuEvents, error) {
-	e, err := bpf.NewPerCpuEvents(k.perfConfig, k.log)
-	if err != nil {
-		return nil, fmt.Errorf("failed kprobe events NewPerCpuEvents: %w", err)
-	}
-	return e, nil
-}
-
-func (k *Observer) observerLost(msg *bpf.PerfEventLost, cpu int) {
-	k.lostCntr++
-}
-
-func (k *Observer) observerError(msg *bpf.PerfEvent) {
-	k.errorCntr++
-}
-
-func isCtxDone(ctx context.Context) bool {
-	select {
-	case <-ctx.Done():
-		return true
-	default:
-		return false
-	}
-}
-
-func (k *Observer) __loopEvents(stopCtx context.Context, e *bpf.PerCpuEvents) error {
-	receiveEvent := func(msg *bpf.PerfEventSample, cpu int) { k.receiveEvent(msg.DataDirect(), cpu) }
-	observerLost := k.observerLost
-	observerError := k.observerError
-	pollTimeoutMsec := int(pollTimeout / time.Millisecond)
-
-	k.log.Info("Listening for events...")
-	k.observerListeners(&readyapi.MsgTetragonReady{})
-
-	for !isCtxDone(stopCtx) {
-		_, err := e.Poll(pollTimeoutMsec)
-		switch {
-		case isCtxDone(stopCtx):
-			k.log.Debug("Context cancelled inside __loopEvents")
-			return nil
-
-		case errors.Is(err, syscall.EBADF):
-			return fmt.Errorf("kprobe events syscall.EBADF: %w", err)
-
-		case err != nil:
-			k.log.WithError(err).Debug("kprobe events poll failed")
-			continue
-		}
-
-		if err := e.ReadAll(maxEventsPerRing, receiveEvent, observerLost, observerError); err != nil {
-			k.log.WithError(err).Warn("kprobe events read failed")
-		}
-
-		ringbufmetrics.ReceivedSet(float64(k.recvCntr))
-		ringbufmetrics.LostSet(float64(k.lostCntr))
-		ringbufmetrics.ErrorsSet(float64(k.errorCntr))
-	}
-	return nil
-}
-
-func (k *Observer) runEvents(stopCtx context.Context) error {
-	/* Probe runtime configuration and do not fail on errors */
-	k.UpdateRuntimeConf()
-
-	e, err := k.__runEvents(stopCtx)
-	if err != nil {
-		return err
-	}
-	defer e.CloseAll()
-
-	k.__loopEvents(stopCtx, e)
-	return nil
-}
-
-func (k *Observer) runEventsNew(stopCtx context.Context, ready func()) error {
-	/* Probe runtime configuration and do not fail on errors */
-	k.UpdateRuntimeConf()
```

Comment:
It looks like this line (added to support Cgroups on older kernels AFAIK) has been lost in the process of swapping over.

Reply:
Yeah we will need to keep this in, IIRC this is part of what resolved that nasty bug on 4.19 with cgroupsv2.

Reply:
ah, nice catch. I messed up the rebase. Let me fix it.

Reply:
should be OK now, PTAL.